scanner/src/main/kotlin/Scanner.kt

```diff
@@ -141,15 +141,72 @@ class Scanner(
             }
         }
 
+        val filesByProvenance = (projectResults.files + packageResults.files)
+            .groupBy { it.provenance }
+            .values
+            .map { fileLists ->
+                fileLists.reduce { acc, next ->
+                    acc.copy(
+                        files = acc.files + next.files
+                    )
+                }
+            }
+            .toSet()
+
+        val deduplicatedScanResults = (projectResults.scanResults + packageResults.scanResults)
+            .groupBy { it.provenance to it.scanner }
```

Comment:
If we have two scan results for the same scanner, shouldn't these be identical. 
Instead of merging the scan results, can we simply drop one of them.

So, analog to what I proposed above, simply use `distinctBy.{ it.provenance to it.scanner }.toSet()` to achive this?

Reply:
> They can have different exclusion rules. Therefore they can be different.

Can you please elaborate a bit in what way exactly? (I don't see how excludes apply here)

Reply:
I verified this and you were right. It makes no difference.